### PR TITLE
Feat(gmx): add options

### PR DESCRIPTION
**Pull request by Quazia**

### CODE REVIEW

**Review by mmackz — COMMENTED** [review]
Do we want this to be explicit?

File: packages/gmx/src/test-setup.ts [hunk 1]
```diff
@@ -105,3 +107,53 @@ export const failingTestCasesV2 = [
   }),
   createTestCase(REDUCE_ORDER_V2, 'when wrong OrderType is used'),
 ]
+
+export const passingOptionsTestCases = [
+  createTestCase(MARKET_SHORT_ETH_ORDER_V2, 'when using a market order'),
+  createTestCase(LIMIT_LONG_ARB_ORDER_V2, 'when using a limit order'),
```
Comment (mmackz) [inline packages/gmx/src/test-setup.ts:113]: would also suggest running some tests with some or all of the parameters undefined ie: any token quests

check line 74 for example

File: .gitignore [hunk 2]
```diff
@@ -10,6 +10,7 @@ coverage
 dist
 node_modules
 tsconfig*.tsbuildinfo
+~
```
Comment (mmackz) [inline .gitignore:13]: curious what this is for

File: .vscode/settings.json [hunk 3]
```diff
@@ -4,8 +4,8 @@
   "typescript.tsdk": "node_modules/typescript/lib",
   "typescript.enablePromptUseWorkspaceTsdk": true,
   "editor.codeActionsOnSave": {
-    "quickfix.rome": true,
```
Comment (mmackz) [inline .vscode/settings.json:7]: do we want this changed?

**Review by mmackz — APPROVED** [review]

File: packages/gmx/src/GMX.ts [hunk 4]
```diff
@@ -90,6 +92,52 @@ export const swap = async (
...
+    },
+  })
+}
+
+export const getOrderType = (orderType: BoostOrderType | undefined) => {
+  switch (orderType) {
+    case BoostOrderType.Market:
+      return {
+        $or: [
+          { orderType: OrderType.MarketSwap },
```
[the remark this answers is not in the document]
Comment (Quazia) [reply packages/gmx/src/GMX.ts:123]: I think we're going to have less support pressure if we just accept it for the time being... we can remove it if it becomes an issue wdyt?

**Comment by Quazia** [comment]
> Do we want this to be explicit?

What explicit?

File: .vscode/settings.json (hunk 3 above)
Comment (Quazia) [reply .vscode/settings.json:7]: Yeah, I just didn't want that change tucked into an outside contributors changeset.

File: .gitignore (hunk 2 above)
Comment (Quazia) [reply .gitignore:13]: Related to pnpm files getting loaded locally - didn't want to push them to remote.

**Comment by mmackz** [comment]
> > Do we want this to be explicit?
> 
> What explicit?



> > Do we want this to be explicit?
> 
> What explicit?

the vscode setttings json

**Review by mmackz — APPROVED** [review]

**Comment by mmackz** [comment]
LGTM

**Pull request merged**